tests/ensemble_evaluator/test_nfs_adaptor.py

```diff
@@ -0,0 +1,56 @@
+import asyncio
+from pathlib import Path
+
+import aiofiles
+import pytest
+import websockets
+from ert_shared.ensemble_evaluator.nsf_adaptor import END_OF_TRANSMISSION, NFSAdaptor
+
+
+async def mock_writer(filename, times=2):
+    async with aiofiles.open(filename, mode="a") as f:
+        for r in range(0, times):
+            await f.write(f"Time {r}\n")
+            await asyncio.sleep(0.2)
+        await f.write(END_OF_TRANSMISSION)
+
+
+async def mock_ws(host, port, done):
+    lines = []
+
+    async def _handler(websocket, path):
+        while True:
+            line = await websocket.recv()
+            lines.append(line)
+            if line == END_OF_TRANSMISSION:
+                done.set_result(lines[:-1])
+                return
+
+    async with websockets.serve(_handler, host, port):
+        await done
+
+
+@pytest.mark.asyncio
+async def test_append_to_file(tmpdir):
+    host = "localhost"
+    port = 50001
+    log_file = Path(tmpdir) / "log"
+    result = asyncio.get_event_loop().create_future()
```

Comment:
Do we need to explicitly create the future? Could we not just return the results from `mock_ws`, since that gives a future anyways?

Reply:
Yeah, good point

Reply:
Still need a future to communicate to the server that we're done. But I don't need it outside the scope of the `mock_ws`. Do you agree?